closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -580,17 +620,22 @@ int decryptFinal(ByteBuffer src, ByteBuffer dst)
 
     // return tag length in bytes
     int getTagLen() {
-
         return this.tagLenBytes;
     }
 
     int getBufferedLength() {
-        if (ibuffer != null && decrypting) {
-            return ibuffer.size();
-        }
+        synchronized (this) {
+            if (decrypting) {
+                if(ibuffer != null) {
+                    return ibuffer.size();
+                }
+            }
 
-        if (ibuffer_enc != null && !decrypting) {
-            return ibuffer_enc.size();
+            if (!decrypting) {
```

Comment:
There's no need to test `decrypting` twice; this can be the `else` block of the `if` statement above.

Reply:
makes sense, I changed it.